### DISCUSSION TITLE
Transfer data between actions

### DISCUSSION THREAD

**Discussion by mambax** [community/community#24973] [Actions]
<p>Before I ask on Stackoverflow:</p>
<p>How does one transfer context between <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idsteps" rel="noopener nofollow ugc">steps</a>?</p>
<p>I see no official guide on that. I see input and output variables in the metadata syntax for Actions.</p>
<p>Can someone point out how to wire them together?</p>
<p>Also addressing the same question is - <a href="https://github.com/lots0logs/gh-action-get-changed-files" rel="noopener nofollow ugc">this action</a> writes to <code>${HOME}</code>. But what is <code>${HOME}</code> and how can I access this in the next job? I asked on the repo but I would like to understand the general concepts around it.</p>
<p>If I write to files, what is the best practice path? For output and for input?</p>
<p>Imagine this:</p>
<pre><code class="lang-auto">    steps:
      - uses: actions/checkout@v2
        with:
          fetch-depth: 0
      - uses: lots0logs/gh-action-get-changed-files@2.1.4
        with:
          token: ${{ secrets.GITHUB_TOKEN }}
      - run: mv ~/files.json . # &lt;--- very uncool
      - uses: coresystemsFSM/actions/checkmarx-check@main # &lt;--- Will need versioning
        with:
          touched-files-file: /github/workspace/files.json &lt;--- also ugly
</code></pre>
<p>How would I properly map this?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>In most cases it’s fine to just write to somewhere within <a href="https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context"><code>github.workspace</code></a>. That’s the default working directory for steps, so you can just use relative paths.</p>
<p>To me the output to <code>$HOME</code> looks like bad design in the <code>lots0logs/gh-action-get-changed-files</code> action, which would make me question if it’s a good idea to use that. If you want to keep using it and it doesn’t have an option to change the output, moving the output file to the workspace is probably the best available solution.</p>
